types/context.go

```diff
@@ -39,8 +40,18 @@ type Context struct {
 	consParams    *tmproto.ConsensusParams
 	eventManager  *EventManager
 	priority      int64 // The tx priority, only relevant in CheckTx
+
+	txBlockingChannels		MessageAccessOpsChannelMapping
+	txCompletionChannels	MessageAccessOpsChannelMapping
+	messageIndex int	// Used to track current message being processed
```

Comment:
why are we tracking message index instead of txindex? in the other places for dependency mapping, we're indexing by the tx index, this distinction is relevant since each tx can have multiple messages as part of it.